highfidelity/hifi
# 2017-10-18: Commerce frontend updates

A few changes here:
1. Add a `showDebugButtons` switch in the QML which dictates whether or not the Debug buttons show up on the "Help" tab of the Wallet
2. Fix "Recent Activity" indentation
3. Fixed login button JavaScript injection on an individual item page when not logged in
4. Small update to Help text for one answer
5. When clicking on "Show Backup Instructions" on the Security page, show the "Instructions Opened on Desktop" popup in Disabled Gray
6. Modify LowerKeyboard button location and image
7. Add "What does 'Rez' mean?" link in Checkout

# Test Plan

## Prerequisites

Currently, you must manually enable "commerce" by adding `"commerce": true,` in your Interface.json (with Interface not running), or by opening Interface Javascript console and evaluating `Settings.setValue("commerce", true)` and restarting.

If your wallet is already set up OR if you have previously set up a wallet on a different build (and thus you have a record of your wallet on the backend), you must perform the following steps to rest your wallet:
1. Open the Wallet app. If you're just resetting your wallet because there's a record of a previous wallet on the backend, go through setup.
2. Go to the "Help" tab in the Wallet app
3. Press the red "DBG: RST Wallet" button at the top of the window

If your wallet is set up but broken (e.g., from testing something that didn't work) such that you are prompted for a passphrase that is never accepted, you need to:
1. Quit interface and delete `%appdata%/Interface <build #>/<username.hifikey>`
2. Restart Interface, set up wallet, and then press the red `DBG: RST WALLET` button as above. Now you are in a truly clean state for wallet setup.

## Tests
Much of the changes here are tiny layout changes or invisible. These changes don't necessitate testing. For the others...

1. Go to your Sandbox
2. Set up your Wallet
3. Log out of High Fidelity
4. Open the Marketplace
5. Ensure you see a green "Log In" banner toward the top of the Marketplace page
6. Click on an individual item (the first one you see is fine) to navigate to the individual item's page
7. Ensure you see a green "Log In" banner toward the top of the page
8. Click the "Log In" button and log in
9. Search the Marketplace for "Test Beach Ball"
10. Buy the Test Beach ball and wait on the confirmation page
11. Ensure you see a "What does 'Rez' mean?" link below the red Rez button
12. Click the "What does 'Rez' mean?" link
13. Verify that you see the "Introduction to Purchases" tutorial. Go through the 2-step tutorial.
14. When you finish the tutorial, verify that you return to the Purchase Confirmation page.
15. Navigate to My Purchases
16. Verify that, upon navigating to My Purchases, you _do not_ see the "Introduction to Purchases" tutorial.
17. Don your HMD
18. Open your Wallet, then navigate to the Security tab. Tap to change your passphrase.
19. When you see the virtual keyboard, verify that you see a "Lower Keyboard" button at the top right of the keyboard. Verify that the button works to lower the keyboard.
20. Cancel this screen and navigate back to the Security tab of the Wallet
21. Click the "View Backup Instructions" link, and verify that the button changes to a disabled button with the text "Instructions Opened on Desktop" for a couple of seconds.
